app/models/vm_or_template/operations/configuration.rb

```diff
@@ -73,8 +73,9 @@ def disconnect_floppies
 
   def raw_add_disk(disk_name, disk_size_mb, options = {})
     raise _("VM has no EMS, unable to add disk") unless ext_management_system
-    if options[:datastore]
-      datastore = Storage.find_by(:name => options[:datastore])
+    if options[:datastore] || options[:datastore_id]
+      datastore = Storage.find(:name => options[:datastore_id]) if options[:datastore_id]
```

Comment:
This is a mistake. I pulled the same code which is common ovirt-provider and automation to Storage and covered with tests.